discovery_engine/test/ffi/ffi_test.dart

```diff
@@ -59,6 +59,7 @@ void main() {
           engineState: null,
           history: [],
           aiConfig: null,
+          sources: {},
```

Comment:
This probably shouldn't be called `sources` but `favoriteSources`. (also in other places)

We do not need to use the exact naming of the http parameter in newscatcher, I mean 
we also don't call `excludedSources` `notSources` :wink:

Reply:
>We do not need to use the exact naming of the http parameter in newscatcher, I mean
we also don't call `excludedSources` `notSources`

true but I felt that `Sources` vs `ExcludedSources` was enough to disambiguate, and is consistent in the same way newscatcher has `sources` vs `not_sources` (even if we don't necessarily copy their exact naming as you mentioned).

but I can change it if you disagree (opinions from others also welcome!)